python/GangaLHCb/Lib/GaudiRunApp/GaudiRunAppHandlers.py

```diff
@@ -0,0 +1,354 @@
+
+from Ganga.GPIDev.Adapters.IRuntimeHandler import IRuntimeHandler
+
+from Ganga.GPIDev.Lib.File.File import File, ShareDir
+from Ganga.GPIDev.Lib.File.FileBuffer import FileBuffer
+from Ganga.Core import ApplicationConfigurationError
+
+from Ganga.Utility.logging import getLogger
+
+from Ganga.GPIDev.Base.Proxy import getName
+
+import os
+import shutil
+from datetime import datetime
+import tempfile
+import tarfile
+import random
+import threading
+from Ganga.Utility.files import expandfilename
+
+from Ganga.GPIDev.Adapters.ApplicationRuntimeHandlers import allHandlers
+
+from GangaGaudi.Lib.RTHandlers.RunTimeHandlerUtils import master_sandbox_prepare, sandbox_prepare, script_generator
+from GangaDirac.Lib.RTHandlers.DiracRTHUtils import dirac_inputdata, dirac_ouputdata, mangle_job_name, diracAPI_script_template, diracAPI_script_settings, API_nullifier, dirac_outputfile_jdl
+from Ganga.GPIDev.Lib.File.OutputFileManager import getWNCodeForOutputPostprocessing
+from Ganga.GPIDev.Adapters.StandardJobConfig import StandardJobConfig
+from Ganga.Utility.util import unique
+from Ganga.Utility.Config import getConfig
+
+from Ganga.GPIDev.Lib.File.LocalFile import LocalFile
+from GangaDirac.Lib.Files.DiracFile import DiracFile
+
+logger = getLogger()
+
+prep_lock = threading.Lock()
+
+def add_timeStampFile(given_path):
+    """
+    This creates a file in this directory given called __timestamp__ which contains the time so that the final file is unique
+    I also add 20 unique characters from the ascii and digits pool from SystemRandom which may reduce the risk of collisions between users
+    Args:
+        given_path (str): Path which we want to create the timestamp within
+    """
+    fmt = '%Y-%m-%d-%H-%M-%S'
+    time_filename = os.path.join(given_path, '__timestamp__')
+    logger.info("Constructing: %s" % time_filename)
+    with open(time_filename, 'a+') as time_file:
+        time_file.write(datetime.now().strftime(fmt))
+        time_file.write('\n'+''.join(random.SystemRandom().choice(string.ascii_uppercase + string.digits) for _ in range(20)))
```

Comment:
uuid is machine dependent, collisions can happen if 2 users are on the same machine, SystemRandom uses the uuid behind the scenes doesn't it?